modules/lazyload/module.php

```diff
@@ -156,9 +192,31 @@ public function __construct() {
 		add_action( 'wp_footer', function() {
 			?>
 			<script type='text/javascript'>
+				const getReapetersBackgroundLazyload = ( lazyloadBackgrounds, reapterAttribute, dataAttribute) => {
+					const lazyloadRepeaterBackgrounds = document.querySelectorAll( `[${ reapterAttribute }]` );
+					const repeaterBackgrounds = [];
+
+					lazyloadRepeaterBackgrounds.forEach( ( repeater ) => {
+						const lazyloadSelector = repeater.getAttribute( dataAttribute );
+						if ( lazyloadSelector ) {
+							lazyloadBackground = repeater.querySelectorAll( lazyloadSelector );
+							if ( lazyloadBackground.length ) {
+								repeaterBackgrounds.push( ...lazyloadBackground );
+							}
+						}
+					} );
+					lazyloadBackgrounds = [ ...lazyloadBackgrounds, ...repeaterBackgrounds ];
+
+					return lazyloadBackgrounds;
+				};
+
 				const lazyloadRunObserver = () => {
 					const dataAttribute = 'data-e-bg-lazyload';
-					const lazyloadBackgrounds = document.querySelectorAll( `[${ dataAttribute }]:not(.lazyloaded)` );
+					const reapterAttribute = 'data-e-bg-repeater';
+
+					let lazyloadBackgrounds = document.querySelectorAll( `[${ dataAttribute }]:not(.lazyloaded, [${reapterAttribute}])` );
+					lazyloadBackgrounds = getReapetersBackgroundLazyload( lazyloadBackgrounds, reapterAttribute, dataAttribute );
```

Comment:
try make it simpler by only observe selectors without the need to know if it is repeater or not

Reply:
I pushed a POC version, please take a look if this is simpler enough. I used like you offered to use simple CSS selectors without knowing which is what on the client, Also in the backend we have now arrayed the selectors for each wrapper.